### PR TITLE
d: gdc: Implement proper dependency gen support

### PR DESCRIPTION
**Pull request by ximion**
Older versions of GDC had broken support for the -MQ flag (resulting in
a compiler error). New versions of GDC have this issue fixed (some older
versions received a backport). Starting with GCC 7.1 we can assume
dependency generation to always work reliably.

### CODE REVIEW

File: mesonbuild/compilers/d.py [hunk 1]
```diff
@@ -233,13 +233,20 @@ def __init__(self, exelist, version, is_cross, **kwargs):
                           '3': default_warn_args + ['-Wextra', '-Wpedantic']}
         self.base_options = ['b_colorout', 'b_sanitize', 'b_staticpic']
 
+        self._has_color_support = version_compare(self.version, '>=4.9')
+        # dependencies were implemented before, but broken - support was fixed in GCC 7.1+
+        # (and some backported versions)
+        self._has_deps_support  = version_compare(self.version, '>=7.1')
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/d.py]: __[Flake8]__

[E221] multiple spaces before operator

**Comment by nirbheek** [comment]
Should this go into the 0.44.1 stable release too?

**Comment by ximion** [comment]
@nirbheek It's not a critical bugfix, but a pretty significant quality-of-life improvement with a close to zero regression chance, so I would say yes, this would be nice to have in 0.44.1.
(With D's heavy use of templating, having the dependency gen not work sometimes causes really weird and unexpected results if you aren't aware of that issue - so fixing it at least for GDC is great. For DMD and LDC there is still a patch pending to be reviewed again and merged, unfortunately, so we can't do much from the Meson side for those compilers yet. See https://github.com/dlang/dmd/pull/6961 )

**Comment by nirbheek** [comment]
Okay, let's try to get this into the stable release once @jpakkane merges it.

**Pull request merged**